json.go

```diff
@@ -0,0 +1,687 @@
+			jsarr := JSONArray{dpiJsonArray: C.dpiData_getJsonArray(&(d.dpiData))}
+			ua, err := jsarr.GetValue()
+			if err != nil {
+				return nil, err
+			}
+			nodes = append(nodes, ua)
+		} else {
+			if elts[i].oracleTypeNum == C.DPI_ORACLE_TYPE_VARCHAR {
+				keyval, err := getJSONScalarString(d)
+				if err == nil {
+					nodes = append(nodes, keyval)
+				} else {
+					return nil, err
+				}
+			} else if elts[i].oracleTypeNum == C.DPI_ORACLE_TYPE_NUMBER {
+				nodes = append(nodes, getJSONScalarNumber(d))
+			} else {
+				nodes = append(nodes, d.Get())
+			}
+		}
+	}
+	return nodes, nil
+}
+
+// It represents the map input.
+type JSONObject struct {
+	dpiJsonObject *C.dpiJsonObject
+}
+
+func (j JSONObject) Len() int { return int(j.dpiJsonObject.numFields) }
+func (j JSONObject) Get() map[string]Data {
+	n := int(j.dpiJsonObject.numFields)
+	names := ((*[maxArraySize]*C.char)(unsafe.Pointer(j.dpiJsonObject.fieldNames)))[:n:n]
+	nameLengths := ((*[maxArraySize]C.uint32_t)(unsafe.Pointer(j.dpiJsonObject.fieldNameLengths)))[:n:n]
+	fields := ((*[maxArraySize]C.dpiJsonNode)(unsafe.Pointer(j.dpiJsonObject.fields)))[:n:n]
+	m := make(map[string]Data, n)
+	for i := 0; i < n; i++ {
+		var d Data
+		jsonNodeToData(&d, &fields[i])
+		m[C.GoStringN(names[i], C.int(nameLengths[i]))] = d
+	}
+	return m
+}
+
+// Returns the Go type map[string]interface{} from JSONObject
+func (j JSONObject) GetValue() (m map[string]interface{}, err error) {
+	m = make(map[string]interface{})
+	n := int(j.dpiJsonObject.numFields)
+	names := ((*[maxArraySize]*C.char)(unsafe.Pointer(j.dpiJsonObject.fieldNames)))[:n:n]
+	nameLengths := ((*[maxArraySize]C.uint32_t)(unsafe.Pointer(j.dpiJsonObject.fieldNameLengths)))[:n:n]
+	fields := ((*[maxArraySize]C.dpiJsonNode)(unsafe.Pointer(j.dpiJsonObject.fields)))[:n:n]
+	for i := 0; i < n; i++ {
+		var d Data
+		jsonNodeToData(&d, &fields[i])
+		if d.NativeTypeNum == C.DPI_NATIVE_TYPE_JSON_OBJECT {
+			jsobj := JSONObject{dpiJsonObject: C.dpiData_getJsonObject(&(d.dpiData))}
+			um, err := jsobj.GetValue()
+			if err != nil {
+				return nil, err
+			}
+			m[C.GoStringN(names[i], C.int(nameLengths[i]))] = um
+		} else if d.NativeTypeNum == C.DPI_NATIVE_TYPE_JSON_ARRAY {
+			jsobj := JSONArray{dpiJsonArray: C.dpiData_getJsonArray(&(d.dpiData))}
+			ua, err := jsobj.GetValue()
+			if err != nil {
+				return nil, err
+			}
+			m[C.GoStringN(names[i], C.int(nameLengths[i]))] = ua
+		} else if fields[i].oracleTypeNum == C.DPI_ORACLE_TYPE_VARCHAR {
+			keyval, err := getJSONScalarString(d)
+			if err == nil {
+				m[C.GoStringN(names[i], C.int(nameLengths[i]))] = keyval
+			} else {
+				return nil, err
+			}
+		} else if fields[i].oracleTypeNum == C.DPI_ORACLE_TYPE_NUMBER {
+			m[C.GoStringN(names[i], C.int(nameLengths[i]))] = getJSONScalarNumber(d)
+		} else {
+			m[C.GoStringN(names[i], C.int(nameLengths[i]))] = d.Get()
+		}
+	}
+	return m, nil
+}
+
+// It populates the fields of struct taking pointer to struct.
```

Comment:
"GetInto populates the fields ..."
Start the method's comment with the method's name.

Reply:
used the function names. Followed the same convention for documenting other functions too.